mlflow/deployments/base.py

```diff
@@ -0,0 +1,168 @@
+"""
+Here it shows the base implementation needed for plugin developers to develop deployment plugins.
+The only interfaces a deployment plugins should expose are the three functions given below
+
+1. :py:func:`get_deploy_client`
+2. :py:func:`run_local`
+3. :py:func:`target_help`
+"""
+
+import abc
+
+
+def run_local(model_uri, flavor=None, config=None):
+    """
+    Deploys the specified model locally, for testing. This method should be defined within the module
+    specified by the plugin author.
+
+    :param model_uri: URI of model to deploy
+    :param flavor: (optional) Model flavor to deploy. If unspecified, a default flavor will be chosen.
+    :param config: (optional) Dict containing updated target-specific configuration for the deployment
+    :return: None
+    """
+    raise NotImplementedError("This function should be implemented in the deployment plugin. It is"
+                              "kept here only for documentation purpose and shouldn't be used in"
+                              "your application")
+
+
+def target_help():
+    """
+    Return a string containing detailed documentation on the current deployment target, to be displayed
+    when users invoke the ``mlflow deployments help -t <target-name>`` CLI. This method should be defined
+    within the module specified by the plugin author.
+    The string should contain:
+    * An explanation of target-specific fields in the ``config`` passed to ``create_deployment``,
+      ``update_deployment``
+    * How to specify a ``target_uri`` (e.g. for AWS SageMaker, ``target_uri``s have a scheme of
+      "sagemaker://<aws-cli-profile-name>", where aws-cli-profile-name is the name of an AWS
+      CLI profile https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html)
+    * Any other target-specific details.
+    """
+    raise NotImplementedError("This function should be implemented in the deployment plugin. It is"
+                              "kept here only for documentation purpose and shouldn't be used in"
+                              "your application")
+
+
+def get_deploy_client(target_uri):
```

Comment:
``get_deploy_client`` actually should not be defined by plugins - instead, we should look up & construct the appropriate subclass of ``BaseDeploymentClient`` based on the provided target URI (e.g. if the scheme of the URI is `redisai`, delegate to the Sagemaker plugin, etc). See the pattern used in the [ArtifactRepositoryRegistry](https://github.com/mlflow/mlflow/blob/1eefa145d62c64cc253b2ba3108bbcbbb852eabc/mlflow/store/artifact/artifact_repository_registry.py#L52-L71) for example